backend/streetdrop-api/src/main/java/com/depromeet/domains/music/genre/service/GenreService.java

```diff
@@ -35,10 +35,12 @@ public Genre getOrCreateGenre(String name) {
 		return genreRepository.save(genre);
 	}
 
-	private SongGenre createSongGenre(Genre genre) {
-		SongGenre songGenre = SongGenre.builder()
-				.genre(genre)
-				.build();
+	private SongGenre getOrCreateSongGenre(Genre genre) {
+		SongGenre songGenre = songGenreRepository.findSongGenreByGenre(genre)
+				.orElseGet(() -> SongGenre.builder()
+						.genre(genre)
+						.build()
+				);
 
 		return songGenreRepository.save(songGenre);
 	}
```

Comment:
```suggestion
	private SongGenre getOrCreateSongGenre(Genre genre) {
		var songGenre = songGenreRepository.findSongGenreByGenre(genre);
		
		if(songGenre.isPresent()){
		     return songGenre;
		}

		return songGenreRepository.save(
		           SongGenre.builder()
						.genre(genre)
						.build()
				));
	}
```

JPA 캐싱을 통한 처리 작업 로직이긴 하지만, 객체를 덜 만들고, 캐싱 접근을 덜 진행하는 방향이 좋지 않을까..
물론, 이게 서버에 큰 영향을 주는 것은 아니긴 해요.

Reply:
귀한 시간 내 주셔서 감사합니다 👍
제안해 주신 코드가 가독성 측면에서 더 좋은 것 같아요~!!

다만 "객체를 덜 만들고, 캐싱 접근을 덜 진행하는 방향"이 이해가 잘 안 되는데, 혹시 조금 더 설명해 주실 수 있을까요?! 🤔

Reply:
@siyeonSon 

객체를 덜 만든 다는 내용은 좀 잘 못 된 것 같고... 메모리에 덜 올라간다가 맞을 것 같긴하네요.

```java
private SongGenre getOrCreateSongGenre(Genre genre) {
		SongGenre songGenre = songGenreRepository.findSongGenreByGenre(genre)
				.orElseGet(() -> SongGenre.builder()
						.genre(genre)
						.build()
				);

		return songGenreRepository.save(songGenre);
	}
```

여기서는 플로우가
1. db에 접근해서 데이터를 가져온다. 만약 데이터가 없다면, SongGenre를 생성한다.
2. JPA의 더티체킹을 통해, 변경지점을 파악하여 데이터를 반환 혹은 save 후 반환을 진행한다.

요렇게 인데.. 무조건 1,2 번의 로직을 타야될까? 라는 생각이 드는 것 같아요.
더티체킹을 갈 필요 없이, if문으로 끝내버리면 뒤에 있을 로직을 타지 않아서 무의미하긴 하지만 로직상 더 좋긴 하거든요. 
(물론, 성능상 이점이 큰건 아닙니다..)

너무 주절이.. 답을 드리는 것 같은데.. 결론은 **더티체킹을 매호출마다 진행해야 될까?** 가 가장 큰 이유입니다.